File: Euclid/Messages/Incoming/UPDATEFLAT.cs
using Euclid.Game;
using Euclid.Network.Streams;
using Euclid.Storage.Database.Access;
using Euclid.Storage.Database.Data;

namespace Euclid.Messages.Incoming
{
    class UPDATEFLAT : IMessageEvent
    {
        public void Handle(Player player, Request request)
        {
            var roomId = int.Parse(request.GetArgument(1, "/"));
            var room = RoomManager.Instance.GetRoom(roomId);

            if (room == null || !room.IsOwner(player.Details.Id))
            {
                return;
            }

            string name = request.GetArgument(2, "/");
            string accessType = request.GetArgument(3, "/");
            bool showOwnerName = request.GetArgument(4, "/") == "1";

            // Name length check
            if (name.Length < 2)
                name = room.Data.Name;

            var state = RoomStatus.OPEN;

            if (accessType == "closed")
                state = RoomStatus.CLOSED;

            if (accessType == "password")
                state = RoomStatus.PASSWORD;

            room.Data.Name = name;
            room.Data.AccessType = state;
            room.Data.ShowName = showOwnerName;

            RoomDao.SaveRoom(room.Data);
        }
    }
}

File: euclid/src/main/java/euclid/messages/incoming/UPDATEFLAT.java
package euclid.messages.incoming;

import euclid.game.Player;
import euclid.game.room.Room;
import euclid.game.room.RoomManager;
import euclid.messages.IMessageEvent;
import euclid.network.streams.Request;
import euclid.storage.database.access.RoomDao;
import euclid.storage.database.data.RoomStatus;

public class UPDATEFLAT implements IMessageEvent {

    @Override
    public void handle(Player player, Request request) {
        int roomId = Integer.parseInt(request.getArgument(1, "/"));
        Room room = RoomManager.getInstance().getRoom(roomId);

        if (room == null || !room.isOwner(player.getDetails().getId())) {
            return;
        }

        String name = request.getArgument(2, "/");
        String accessType = request.getArgument(3, "/");
        boolean showOwnerName = request.getArgument(4, "/").equals("1");

        if (name.length() < 2) {
            name = room.getData().getName();
        }

        RoomStatus state = RoomStatus.OPEN;

        if (accessType.equals("closed")) {
            state = RoomStatus.CLOSED;
        }

        if (accessType.equals("password")) {
            state = RoomStatus.PASSWORD;
        }

        room.getData().setName(name);
        room.getData().setAccessType(state);
        room.getData().setShowName(showOwnerName);

        RoomDao.saveRoom(room.getData());
    }
}
